experimental/dds/tree2/src/feature-libraries/sequence-field/sequenceFieldToDelta.ts

```diff
@@ -33,57 +33,55 @@ export function sequenceFieldToDelta<TNodeChange>(
-			if (deltaMark.fields.size === 0) {
-				delete deltaMark.fields;
+			const nestedDelta = deltaFromChild(changes);
+			if (nestedDelta.size > 0) {
+				if (inputCellId === undefined) {
+					deltaMark.fields = nestedDelta;
+				} else {
+					global.push({
+						id: nodeIdFromChangeAtom(inputCellId),
+						fields: nestedDelta,
+					});
+				}
 			}
 		}
 		if (!areInputCellsEmpty(mark) && !areOutputCellsEmpty(mark)) {
 			// Since each cell is associated with exactly one node,
 			// the cell starting end ending populated means the cell content has not changed.
 			local.push(deltaMark);
-		} else if (areInputCellsEmpty(mark) && areOutputCellsEmpty(mark)) {
-			const inputCellId = getInputCellId(mark, revision, undefined);
+		} else if (isTransientEffect(mark)) {
 			assert(inputCellId !== undefined, "Transient mark should have defined input cell ID");
 			// The cell starting and ending empty means the cell content has not changed,
 			// unless transient content was inserted/attached.
-			if (isTransientEffect(mark)) {
-				if (isMoveDestination(mark.attach) && isMoveSource(mark.detach)) {
-					assert(mark.changes === undefined, "Transient moves should not have changes");
-					continue;
-				}
+			if (isMoveDestination(mark.attach) && isMoveSource(mark.detach)) {
+				assert(mark.changes === undefined, "Transient moves should not have changes");
+				continue;
+			}
 
-				const outputId = getOutputCellId(mark, revision, undefined);
-				assert(outputId !== undefined, "Transient mark should have defined output cell ID");
-				const oldId = nodeIdFromChangeAtom(
-					isMoveDestination(mark.attach)
-						? getEndpoint(mark.attach, revision)
-						: inputCellId,
-				);
-				if (!areEqualChangeAtomIds(inputCellId, outputId)) {
-					if (mark.attach.type === "Insert" && mark.attach.content !== undefined) {
-						build.push({
-							id: oldId,
-							trees: mark.attach.content.map(cursorForJsonableTreeNode),
-						});
-					}
-					rename.push({
-						count: mark.count,
-						oldId,
-						newId: nodeIdFromChangeAtom(outputId),
-					});
-				}
-				if (deltaMark.fields) {
-					global.push({
+			const outputId = getOutputCellId(mark, revision, undefined);
+			assert(outputId !== undefined, "Transient mark should have defined output cell ID");
+			const oldId = nodeIdFromChangeAtom(
+				isMoveDestination(mark.attach) ? getEndpoint(mark.attach, revision) : inputCellId,
+			);
+			if (!areEqualChangeAtomIds(inputCellId, outputId)) {
+				if (mark.attach.type === "Insert" && mark.attach.content !== undefined) {
+					build.push({
 						id: oldId,
-						fields: deltaMark.fields,
+						trees: mark.attach.content.map(cursorForJsonableTreeNode),
 					});
 				}
-			} else if (deltaMark.fields) {
+				rename.push({
+					count: mark.count,
+					oldId,
+					newId: nodeIdFromChangeAtom(outputId),
+				});
+			}
+			if (deltaMark.fields) {
 				global.push({
-					id: nodeIdFromChangeAtom(inputCellId),
+					id: oldId,
```

Comment:
I'm realizing I'm a bit confused about which ID the changes for a node should be listed under.

Reply:
All delta changes are listed in the input-context location of the node they affect.